File: OsmAnd/src/net/osmand/plus/api/SQLiteAPI.java
package net.osmand.plus.api;



public interface SQLiteAPI {

	public interface SQLiteConnection {
		
		void close();
		
		SQLiteCursor rawQuery(String sql, String[] selectionArgs);

		void execSQL(String query);
		
		void execSQL(String query, Object[] objects);

		SQLiteStatement compileStatement(String string);

		void setVersion(int newVersion);
		
		int getVersion();

		boolean isReadOnly();

		boolean isDbLockedByOtherThreads();

		boolean isClosed();

		void beginTransaction();
		
		void endTransaction();
	}
	
	public interface SQLiteCursor {
		
		String[] getColumnNames();

		boolean moveToFirst();
		
		boolean moveToNext();

		/**
		 * Takes parameter value (zero based)
		 */
		String getString(int ind);

		double getDouble(int ind);
		
		long getLong(int ind);
		
		long getInt(int ind);
		
		byte[] getBlob(int ind);
		
		void close();

		
		
	}
	
	public interface SQLiteStatement {

		// 1 based argument
		void bindString(int i, String filterId);

		// 1 based argument
		void bindNull(int i);

		void execute();

		void close();

		long simpleQueryForLong();

		String simpleQueryForString();

		void bindLong(int i, long val);
		
		void bindBlob(int i, byte[] val);
		
		void clearBindings();

	}
	
	public SQLiteConnection getOrCreateDatabase(String name, boolean readOnly);
	
	public SQLiteConnection openByAbsolutePath(String path, boolean readOnly);
}

File: OsmAnd-shared/src/commonMain/kotlin/net.osmand.shared/api/SQLiteAPI.kt
package net.osmand.shared.api

interface SQLiteAPI {
	interface SQLiteConnection {
		fun close()
		fun rawQuery(sql: String, selectionArgs: List<String>): SQLiteCursor?
		fun execSQL(query: String)
		fun execSQL(query: String, objects: List<Any>)
		fun compileStatement(query: String): SQLiteStatement
		var version: Int
		val isReadOnly: Boolean
		val isClosed: Boolean
	}

	interface SQLiteCursor {
		val columnNames: List<String>
//		fun moveToFirst(): Boolean
		fun moveToNext(): Boolean

		/**
		 * Takes parameter value (zero based)
		 */
		fun getString(index: Int): String?
		fun getDouble(ind: Int): Double?
		fun getLong(ind: Int): Long?
		fun getInt(ind: Int): Int?
		fun getBlob(ind: Int): ByteArray?
		fun isNull(ind: Int): Boolean
		fun getColumnIndex(columnName: String): Int
		fun close()
	}

	interface SQLiteStatement {
		fun bindString(index: Int, value: String)
		fun bindNull(index: Int)
		fun execute()
		fun close()
//		fun simpleQueryForLong(): Long
//		fun simpleQueryForString(): String?
		fun bindLong(index: Int, value: Long)
		fun bindBlob(index: Int, value: ByteArray)
	}

	fun getOrCreateDatabase(name: String, readOnly: Boolean): SQLiteConnection

	fun openByAbsolutePath(path: String, readOnly: Boolean): SQLiteConnection
}
